libgearman-server/plugins/queue/redis/queue.cc

```diff
@@ -153,9 +153,19 @@ bool gearmand::plugins::queue::Hiredis::fetch(char *key, gearmand::plugins::queu
   } else {
     // 2 x (key + value)
     assert(reply->elements == 4);
-    std::string s{reply->element[1]->str};
-    req.data = s;
-    req.priority = (uint32_t)std::stoi(reply->element[3]->str);
+    auto fk = reply->element[0]->str;
+    if(strcmp(fk, "data") == 0) {
+      std::string s{reply->element[1]->str};
+      req.data = s;
+      req.priority = (uint32_t)std::stoi(reply->element[3]->str);
+    } else if (strcmp(fk, "priority") == 0) {
+      std::string s{reply->element[3]->str};
+      req.data = s;
+      req.priority = (uint32_t)std::stoi(reply->element[1]->str);
+    } else {
+      std::string ex{"unexpected key at first position in "};
+      throw std::invalid_argument(ex.append(key));
```

Comment:
We've never thrown exceptions before in any plugins. This throw is going to crash the thread and really all of gearmand. I know this happens only on startup, but it's a pretty nasty way to explode.

I suggest instead we return false, which the code handles well and which will tell the user which key failed.